sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RequestRetryCondition.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.policy;
+
+import com.generic.core.http.models.HttpResponse;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Information about the request that failed, used to determine whether a retry should be attempted.
+ */
+public final class RequestRetryCondition {
+    private final HttpResponse response;
+    private final Throwable throwable;
+    private final int tryCount;
+    private final List<Throwable> retriedExceptions;
+
+    /**
+     * Creates a new ShouldRetryInfo object.
+     *
+     * @param response The HTTP response of the request that failed.
+     * @param throwable The throwable of the request that failed.
+     * @param tryCount The number of tries that have been attempted.
+     * @param retriedThrowables The list of throwables that have been encountered during retries.
+     */
+    RequestRetryCondition(HttpResponse response, Throwable throwable, int tryCount,
+                          List<Throwable> retriedThrowables) {
```

Comment:
For generic-core, since this is net new, let's drop down `Throwable` to `Exception` as we shouldn't be retrying `Error`s.